pykickstart/commands/ntp.py

```diff
@@ -0,0 +1,137 @@
+        self.nts = kwargs.get("nts", False)
+
+    def __eq__(self, y):
+        if not y:
+            return False
+
+        if self.server:
+            return self.server == y.server and self.nts == y.nts
+        elif self.pool:
+            return self.pool == y.pool and self.nts == y.nts
+
+    def __ne__(self, y):
+        return not self == y
+
+    def __str__(self):
+        retval = BaseData.__str__(self)
+
+        args = self._getArgsAsStr()
+        if args:
+            retval += "ntp%s\n" % args
+
+        return retval
+
+    def _getArgsAsStr(self):
+        retval = ""
+
+        if self.server:
+            retval += " --server=%s" % self.server
+        if self.pool:
+            retval += " --pool=%s" % self.pool
+        if self.nts:
+            retval += " --nts"
+
+        return retval
+
+
+class F33_NTP(KickstartCommand):
+    removedKeywords = KickstartCommand.removedKeywords
+    removedAttrs = KickstartCommand.removedAttrs
+
+    def __init__(self, writePriority=0, *args, **kwargs):
+        KickstartCommand.__init__(self, writePriority, *args, **kwargs)
+        self.op = self._getParser()
+        self.NTPList = kwargs.get("NTPList", [])
+        self.exclusive_required_options = [("server", "--server"),
+                                           ("pool", "--pool")]
+
+
+    def __str__(self):
+        retval = ""
+        for ntp_command in self.NTPList:
+            retval += ntp_command.__str__()
+
+        return retval
+
+    def _getParser(self):
+        op = KSOptionParser(prog="ntp", description="""
+                            Configures the Network Time Protocol.""",
+                            version=F33)
+        op.add_argument("--server", version=F33, help="""
+                        A single NTP server.
+
+                        ``--server`` and ``--pool`` are mutually exclusive.
+                        """,
+                        metavar="<group1>,<group2>,...,<groupN>")
+        op.add_argument("--pool", version=F33, help="""
+                        A single NTP pool.
+
+                        ``--pool`` and ``--server`` are mutually exclusive.
+                        """)
+        op.add_argument("--nts", dest="nts", action="store_true",
+                        default=False, version=F33, help="""
+                        If specified, consider the server/pool to be
+                        a NTS server/pool. Without this option it
+                        will be considered to be a plain NTP server/pool
+                        without NTS support.
+                        """)
+        return op
+
+    def parse(self, args):
+        nd = self.dataClass()   # pylint: disable=not-callable
+        ns = self.op.parse_args(args=args, lineno=self.lineno)
+        self.set_to_obj(ns, nd)
+        nd.lineno = self.lineno
+
+        # Check that just one of exclusive required options is specified
+        used_options = [opt for attr, opt in self.exclusive_required_options
```

Comment:
There is nothing like that unfortunately in the KSOptionParser class as far as I can tell. The same code is used in repo and url:

https://github.com/pykickstart/pykickstart/blob/master/pykickstart/commands/repo.py#L279
https://github.com/pykickstart/pykickstart/blob/master/pykickstart/commands/url.py#L189

This is effectively the third copy of it, so it indeed makes sense to de-duplicate it - @bcl any suggestions where to put it in via followup up PR (I think a PR adding a new command should ideally not do backend changes like this one)?

Reply:
I guess options.py would make sense, since it is dealing with options.

Reply:
> I guess options.py would make sense, since it is dealing with options.

OK, will do a followup PR to deduplicate this once this one is merged (so that we don't block implementation of the F33 NTS support change).